src/client/util/zip.h

```diff
@@ -59,44 +66,279 @@ namespace client {
+        m_Data->compressed.resize(m_Data->compressed_size);
+        // Read in compressed data
+        m_Stream->Read(*m_Data->compressed.data(), m_Data->compressed.size());
+        switch (m_Data->compression_method)
+          {
+            case static_cast<std::size_t>(Method::deflate):
+              {
+                LOG(debug) << "ZIP: file uses compression method 'deflate'";
+                // Instantiate decompressor
+                kovri::core::DeflateDecompressor decompressor;
+                // Put in data to decompress
+                decompressor.Put(
+                    m_Data->compressed.data(), m_Data->compressed.size());
+                // Test if uncompressed size will be valid
+                if (decompressor.MaxRetrievable() <= m_Data->uncompressed_size)
+                  {
+                    // Resize for next file
+                    m_Data->uncompressed.resize(m_Data->uncompressed_size);
+                    decompressor.Get(
+                        m_Data->uncompressed.data(),
+                        m_Data->uncompressed.size());
+                    // Verify checksum
+                    if ((decompressor.Verify(
+                            m_Data->crc_32.data(),
+                            m_Data->uncompressed.data(),
+                            m_Data->uncompressed.size())))
+                      {  // Checksum passed
+                        // Store/map the uncompressed file
+                        m_Contents.insert(
+                            {m_Data->local_file_count,
+                             std::move(m_Data->uncompressed)});
+                      }
+                    else
+                      {
+                        LOG(error) << "ZIP: CRC-32 Failed";
+                        return false;
+                      }
+                  }
+                else
+                  {
+                    LOG(error) << "ZIP: actual uncompressed size "
+                               << decompressor.MaxRetrievable() << " exceeds "
+                               << m_Data->uncompressed_size << " from header";
+                    return false;
+                  }
+                break;
+              }
+            case static_cast<std::size_t>(Method::stored):
+              {
+                LOG(debug) << "ZIP: file uses compression method 'stored'";
+                // Store/map the local file as-is
+                m_Contents.insert(
+                    {m_Data->local_file_count, std::move(m_Data->compressed)});
+                break;
+              }
+            default:
+              LOG(error) << "ZIP: file uses an unsupported compression method";
+              return false;
+          }
+      }
+    catch (...)
+      {
+        m_Exception.Dispatch(__func__);
+        return false;
+      }
+    LOG(debug) << "ZIP: successfully processed file";
+    m_Data->local_file_count++;  // Move onto next file for processing
+    return true;
+  }
```

Comment:
I don't agree with moving this all out of the implementation file. Why did you want to do this?